### PR TITLE
SURVEY TIME BABY! Foodsurvey by Emelie Svensson

### PR DESCRIPTION
**Pull request by emeliesv**
Link to my deployed project:
https://xenodochial-johnson-f4667d.netlify.app/

My reflections on how this week's project turned out:
I'm really proud of this survey. I think it has a clean design, it's funny and I worked real hard on the components being reusable.

### CODE REVIEW

File: code/public/index.html [hunk 1]
```diff
@@ -4,31 +4,18 @@
...
-      work correctly both with client-side routing and a non-root public URL.
-      Learn how to configure a non-root public URL by running `npm run build`.
-    -->
-    <title>Technigo React App</title>
+    <meta name="description" content="A fantastic survey about food" />
+    <meta name="keywords" content="Food, Survey, React">
+    <meta property="og:title" content="Food Survey" />
+    <meta property="og:description" content="A fantatstic survey about food" />
+    <meta property="og:url" content="https://xenodochial-johnson-f4667d.netlify.app/" />
+    <title>Food and restaurant survey!</title>
```
Comment (ylvan75) [inline code/public/index.html:12]: Really nice with these meta tags, especially `og:`

File: code/src/components/Button.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+import './button.css'
+
+export const Button = (props) => {
+return <button tabIndex={'0'} type={props.type} onClick={props.click} disabled={props.disable}>{props.text}</button>
```
Comment (ylvan75) [inline code/src/components/Button.js:5]: New for me to write `{'0'}`

File: code/src/components/Checkbox.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+export const Checkbox = (props) => {
+    return (
+        <>
+            <h2 tabIndex={'0'}>{props.question}</h2>
+                <label htmlFor={'checkbox'}>
+                    <input
+                        tabIndex={'0'}
+                        id={'checkbox'}
+                        type="checkbox"
```
Comment (ylvan75) [inline code/src/components/Checkbox.js:11]: Shouldn't this be `{'checkbox'}`, to be consistent?

File: code/src/components/Header.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import React from 'react'
+import './header.css'
+
+export const Header = () => {
+    return(
+        <header tabIndex={'0'} className="header">
```
Comment (ylvan75) [inline code/src/components/Header.js:6]: Shouldn't this be `{'header'}`, to be consistent?

File: code/src/components/Survey.js [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+  const [showSummary, setShowSummary] = useState(false);
+
+  /* This function uses a for-loop to extract the 'text' from each object inside the different 
+  arrays so that the text is displayed in summary and not the value.
+  The value could be used for backend instead (although at the moment it's just not used) */
+  const getUserInput = (userValue, optionsArray) => {
+    for (let i = 0; i < optionsArray.length; i++) {
+      if (optionsArray[i].value === userValue)
+        return optionsArray[i].text
+    } return 'no text found'
```
Comment (ylvan75) [inline code/src/components/Survey.js:32]: Nice solution! What inspired you to do like this?

File: code/src/components/Survey.js [hunk 6]
```diff
@@ -0,0 +1,368 @@
...
+    },
+    {
+      text: "Non-alcoholic fancy cocktail",
+      value: "non-alcoholic",
+    },
+    {
+      text: "TEQUILA BABY",
+      value: "tequila",
+    },
+  ]
```
Comment (ylvan75) [inline code/src/components/Survey.js:60]: Missing semi-colon

File: code/src/components/Survey.js [hunk 7]
```diff
@@ -0,0 +1,368 @@
...
+    },
+    {
+      text: "Big freaking bowl of pasta",
+      value: "pasta",
+    },
+    {
+      text: "Just a salad please",
+      value: "salad",
+    },
+  ]
```
Comment (ylvan75) [inline code/src/components/Survey.js:87]: Missing semi-colon

File: code/src/components/Survey.js [hunk 8]
```diff
@@ -0,0 +1,368 @@
...
+    {
+      text: "Guide Michelin all day baby",
+      value: "fancypants",
+    },
+    {
+      text:
+        "Anything where you can order macrobiotic, nonprocessed and prefereably raw food please",
+      value: "dietician",
+    },
+  ]
```
Comment (ylvan75) [inline code/src/components/Survey.js:116]: Missing semi-colon

File: code/src/components/Survey.js [hunk 9]
```diff
@@ -0,0 +1,368 @@
...
+    },
+    {
+      text: "Broccoli",
+      value: "broccoli",
+    },
+    {
+      text: "Some fruits are always nice.",
+      value: "fruit",
+    },
+  ]
```
Comment (ylvan75) [inline code/src/components/Survey.js:143]: Missing semi-colon

File: code/src/components/Survey.js [hunk 10]
```diff
@@ -0,0 +1,368 @@
...
+                />
+                <div className="button-container">
+                  <Button
+                    type='button'
+                    text='Previous Pretty Please!'
+                    click={previousQuestion}
+                  />
+                  <Button
+                    type='button'
+                    text={celebrationDrink ? 'Next Baby!': 'Choose something!'}
```
Comment (ylvan75) [inline code/src/components/Survey.js:184]: I like this! That you use Conditional (ternary) operator.

File: code/src/components/Survey.js [hunk 11]
```diff
@@ -0,0 +1,368 @@
...
+                    },
+                    {
+                      value: '41-60',
+                      message: 'Prime years baby'
+                    },
+                    {
+                      value: '61-100',
+                      message: 'Oldie but goldie baby'
+                    }
+                  ]}
```
Comment (ylvan75) [inline code/src/components/Survey.js:289]: Couldn't this also be an array with object that you declare in the same way as you did with the other array's?

File: code/src/components/Text.js [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+        <>
+        <h2 tabIndex={'0'}>What did you think about this awesome survey:</h2>
+        <div className="text-input">
+            <input tabIndex={'0'} type="text" onChange={(event) => props.setUserText(event.target.value)}/>
+        </div>
+        </>
+    )
+}
+/* Function for next question button:
+const nextQuestion = () => setQuestion(question + 1) //to get next question in line (add +1 later) const previousQuestion = () => setQuestion(question - 1) */
```
Comment (ylvan75) [inline code/src/components/Text.js:15]: I think you can move this comment to Survey.js where you have declared these functions if you feel that you want to keep the comment, if not I think you can remove it.

File: code/src/components/dropdown.css [hunk 13]
```diff
@@ -0,0 +1,6 @@
+select {
+    max-width: 250px;
+}
+.dropdown {
```
Comment (ylvan75) [inline code/src/components/dropdown.css:4]: Add a line break

File: code/src/components/survey.css [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+.button-container {
+    display: flex;
+    justify-content: space-evenly;
+}
+
+@media screen and (min-width:768px) {
+    .wrapper {
+        padding: 20%
```
Comment (ylvan75) [inline code/src/components/survey.css:27]: Missing semi-colon

**Review by ylvan75 — APPROVED** [review]
Wow Emelie :-)

I liked reading your README file, especially when you say "I'm very proud of how this turned out, since I wanted it to be completely reusable.".

And YES you should be. You have done a great job!

It's really nice that you have the requirements in the README. Made it easy for me to start testing your site.

I took some time and looked at your site with these tools and browsers/devices and what I could see it looked good.

One thing that myself highlighted in my README is that "Screen reader" is tricky and sometimes I feel a little bit lost how to use it and what is the best practice to make sure your site works good. 
 

Accessibility
-Screen reader - Chromevox
-Lighthouse
-https://wave.webaim.org/
-https://validator.w3.org/

Cross-browser
-Chrome - Mac
-Firefox - Windows
-Edge - Windows
-iOS - iPone X (Safari)
-Android - Samsung S8 (Chrome)


**Don't forget to be consistent** if you want to use s**emi-colon**. Survey.js has a mix and in all the smaller components you don't seem to have any semi-colon.

File: code/src/components/Radio.js [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+    )
+
+}
+
+
+
+
+
+
+/* This component is completely reusable since it uses props.
```
Comment (HIPPIEKICK) [inline code/src/components/Radio.js:33]: ⭐

File: code/src/components/Checkbox.js (hunk 3 above)
Comment (HIPPIEKICK) [reply code/src/components/Checkbox.js:11]: I think these could all be simple quotes - either way - as Ylva says, you should be consistent (including consistency regarding double quotes or single quotes)

File: code/src/components/Dropdown.js [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+            <h2 tabIndex={'0'}>{props.question}</h2>
+            <select tabIndex={'0'} onChange={(event) => props.setSelectValue(event.target.value)} value={props.selectValue} required>
+                {props.optionsArray.map((object) => (
+                    <option tabIndex={'0'} className="dropdown" key={object.value} value={object.value}>{object.text}</option>
+                ))}
+            </select>
+        </div>
+    )
+
+}
```
Comment (HIPPIEKICK) [inline code/src/components/Dropdown.js:17]: Semicolons missing

File: code/src/components/Dropdown.js [hunk 17]
```diff
@@ -0,0 +1,17 @@
+import React from 'react'
+import './dropdown.css'
+
+export const Dropdown = (props) => {
+
+    return (
+        <div>
+            <h2 tabIndex={'0'}>{props.question}</h2>
+            <select tabIndex={'0'} onChange={(event) => props.setSelectValue(event.target.value)} value={props.selectValue} required>
```
Comment (HIPPIEKICK) [inline code/src/components/Dropdown.js:9]: New lines for when an element has more than two attributes. Convention to always start with booleans:
```
<select
  required
  tabIndex={'0'}
  onChange={(event) => props.setSelectValue(event.target.value)}
  value={props.selectValue}>
```

File: code/src/components/Radio.js [hunk 18]
```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+import './radio.css'
+
+
+export const Radio = (props) => {
+    console.log(props.ageGroup)
```
Comment (HIPPIEKICK) [inline code/src/components/Radio.js:6]: Remove `console.log`'s before deployment

File: code/src/components/Radio.js [hunk 19]
```diff
@@ -0,0 +1,35 @@
...
+        </>
+    )
+
+}
+
+
+
+
+
+
```
Comment (HIPPIEKICK) [inline code/src/components/Radio.js:32]: It's enough with a single empty line

File: code/src/components/Summary.js [hunk 20]
```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import './summary.css'
+/* This is not completely reusable, if I would like to do so 
+I'd change the text inside each <p></p> to be props and specify the props in Survey.js. */
+export const Summary = (props) => {
```
Comment (HIPPIEKICK) [inline code/src/components/Summary.js:5]: We don't need parentheses when there's only one argument :)

File: code/src/components/Survey.js [hunk 21]
```diff
@@ -0,0 +1,368 @@
...
+  const nextQuestion = () => setQuestion(question + 1);
+  const previousQuestion = () => setQuestion(question - 1);
+  const [ageGroup, setAgeGroup] = useState();
+  const [checkbox, setCheckbox] = useState(false); /* Needs to be specified to either true or false to be controlled from the start */
+  const [celebrationDrink, setCelebrationDrink] = useState('');
+  const [rainyFood, setRainyFood] = useState('');
+  const [restaurantHunt, setRestaurantHunt] = useState('');
+  const [saladMaker, setSaladMaker] = useState('');
+  const [userText, setUserText] = useState();
+  const [showSummary, setShowSummary] = useState(false);
```
Comment (HIPPIEKICK) [inline code/src/components/Survey.js:23]: I would keep all states together and the functions beneath

File: code/src/components/Survey.js [hunk 22]
```diff
@@ -0,0 +1,368 @@
...
+  /* This function uses a for-loop to extract the 'text' from each object inside the different 
+  arrays so that the text is displayed in summary and not the value.
+  The value could be used for backend instead (although at the moment it's just not used) */
+  const getUserInput = (userValue, optionsArray) => {
+    for (let i = 0; i < optionsArray.length; i++) {
+      if (optionsArray[i].value === userValue)
+        return optionsArray[i].text
+    } return 'no text found'
+  }
+  /* Arrays with objects for each components, needs to be definied outside of the component-scope so that getUserInput() can access it. */
```
Comment (HIPPIEKICK) [inline code/src/components/Survey.js:34]: Maybe these arrays could be in their own components to clean up this file a bit

**Review by HIPPIEKICK — CHANGES_REQUESTED** [review]
Really nice implementation! Changing to CR because it's not following code guidelines (consistency with semicolons, indentation)

File: code/src/components/Survey.js (hunk 22 above)
Comment (emeliesv) [reply code/src/components/Survey.js:34]: If I want to move these into a different component, how shouldI specify which array I'd like to use in Survey.js and how can I make sure that getUserInput() can access them?

File: code/src/components/Button.js [hunk 23]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+
+import './button.css'
```
Comment (HIPPIEKICK) [inline code/src/components/Button.js:3]: semicolon ;)

File: code/src/components/Checkbox.js [hunk 24]
```diff
@@ -0,0 +1,18 @@
+import React from 'react'
```
Comment (HIPPIEKICK) [inline code/src/components/Checkbox.js:1]: semicolon

File: code/src/components/Checkbox.js [hunk 25]
```diff
@@ -0,0 +1,18 @@
...
+                    <input
+                        tabIndex={'0'}
+                        id={'checkbox'}
+                        type={'checkbox'}
+                        value={props.answer}
+                        checked={props.checkbox}
+                        onChange={(event) => props.setCheckbox(event.target.checked)}
+                    />{props.answer}</label>
+        </>
+    )
```
Comment (HIPPIEKICK) [inline code/src/components/Checkbox.js:17]: semicolon

File: code/src/components/Survey.js (hunk 22 above)
Comment (HIPPIEKICK) [reply code/src/components/Survey.js:34]: You should just be able to import them as usual:

**drinkArray.js**
`export const drinkArray = [***]`

`import { drinkArray } from './drinkArray'`

File: code/src/components/Checkbox.js (hunk 25 above)
Comment (HIPPIEKICK) [reply code/src/components/Checkbox.js:17]: Semicolons still missing - check all files :)

File: code/src/components/Text.js [hunk 26]
```diff
@@ -0,0 +1,14 @@
...
+
+export const Text = (props) => {
+    return(
+        <>
+        <h2 tabIndex={'0'}>What did you think about this awesome survey:</h2>
+        <div className={'text-input'}>
+            <input tabIndex={'0'} type='text' onChange={(event) => props.setUserText(event.target.value)}/>
+        </div>
+        </>
+    )
```
Comment (HIPPIEKICK) [inline code/src/components/Text.js:13]: 🙈

File: code/src/components/comfortFoodArray.js [hunk 27]
```diff
@@ -0,0 +1,26 @@
+export const comfortFoodArray = [
```
Comment (HIPPIEKICK) [inline code/src/components/comfortFoodArray.js:1]: 😍

**Review by HIPPIEKICK — APPROVED** [review]

File: code/src/components/Text.js (hunk 26 above)
Comment (emeliesv) [reply code/src/components/Text.js]: :octocat:

**Pull request closed without merging**